### PR TITLE
Less string allocations

### PR DESCRIPTION
**Pull request by Bond-009**
Wanted to take the new `Span` for a spin :p

### CODE REVIEW

File: Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs [hunk 1]
```diff
@@ -78,18 +78,37 @@ private static string GetHandlerPathIfAny(string listenerUrl)
...
+                    return remoteIp = temp.ToString();
+                }
+
+                temp = CheckBadChars(XRealIp);
+                if (temp.Length != 0)
+                {
+                    return remoteIp = NormalizeIp(temp).ToString();
+                }
+
+                return remoteIp = NormalizeIp(request.RemoteEndPoint?.Address.ToString()).ToString();
```
Comment (JustAMan) [inline Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: what would happen if `RemoteEndPoint` is `null`, won't we get `remoteIp` set to a string of literal value "null"?
or an exception in `NormalizeIp` that is when it tries to parse IP address string that is just "null"?

Comment (cvium) [reply Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: The first toString is never executed if `RemoteEndPoint` is null. And I'm fairly certain that null.ToString() would throw an exception.

Comment (JustAMan) [reply Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: So if `RemoteEndPoint` is `null` it means this effectively transforms to
```cs
return remoteIp = NormalizeIp(null).ToString();
```
Right?

Comment (Bond-009) [reply Jellyfin.Server/SocketSharp/WebSocketSharpRequest.cs]: @JustAMan Correct

**Comment by joshuaboniface** [comment]
@Bond-009 Merge conflict from a previous PR of yours, please review :-)

**Comment by JustAMan** [comment]
@Bond-009 your PR fails to build in the CI, please look.

**Comment by joshuaboniface** [comment]
The CI is good now, it's failing at a known-failing step. Merging.

**Pull request merged**